source/rhoas/src/main/java/com/openshift/cloud/RHOASOperator.java

```diff
@@ -69,9 +71,43 @@ public int run(String... args) throws Exception {
     config = configuration.getConfigurationFor(serviceRegistryConnectionController);
     LOG.info("CR class: " + config.getCustomResourceClass());
 
+    cleanupDeletedLegacyFinalizers();
+
     operator.start();
 
     Quarkus.waitForExit();
     return 0;
   }
+
+  private void cleanupDeletedLegacyFinalizers() {
+
+    // Clean up legacy Finalizers for deleted resources
+    // We removed finalizers for several types, but didn't correctly remove them from instanced
+    // resources in k8s
+    // if those resources were deleted then we need to remove the finalizers here outside of the
+    // normal reconsiliation loop
+    var srcs = new ArrayList<>(client.serviceRegistryConnection().list().getItems());
+    srcs.removeIf(item -> !item.isMarkedForDeletion());
+    srcs.forEach(
+        src -> src.removeFinalizer("serviceregistryconnections.rhoas.redhat.com/finalizer"));
+    srcs.forEach(src -> client.serviceRegistryConnection().createOrReplace(src));
+
+    var csrs = new ArrayList<>(client.cloudServicesRequest().list().getItems());
+    csrs.removeIf(item -> !item.isMarkedForDeletion());
+    csrs.forEach(csr -> csr.removeFinalizer("cloudservicesrequests.rhoas.redhat.com/finalizer"));
+    csrs.forEach(csr -> client.cloudServicesRequest().createOrReplace(csr));
+
+    var akcs = new ArrayList<>(client.kafkaConnection().list().getItems());
```

Comment:
Do we have permission to do this from operator?
I think our service account has only namespaced account, but if you have tested it it should be fine.

Reply:
If that works get item doesnt have paggination so it will load all objects into memory.
It should not be issue for 100k objects but yes

Reply:
I only tested locally, but I've reached out to the sandbox team to confirm.